### ISSUE TITLE
Support reporting crashes directly to GitLab

### ISSUE THREAD

**Issue by mcatanzaro** [abrt/abrt#1453]
This is a pretty substantial feature request, so I know it might be a tough sell, but on Red Hat Bugzilla we are absolutely overwhelmed with GNOME bug reports, to the degree that I'm firmly convinced we need to start running an automated script to close incoming bug reports and direct users to report directly upstream instead. ABRT bug reports are essential to improving quality, but they'd do a lot more good if they were reported directly upstream, where the right maintainer can look at the report and take action, rather than reporting them downstream where the maintainers are not specialized in the packages they own and do not have time to triage the bugs or forward them upstream.

It would help a lot if ABRT could report crashes directly to GNOME GitLab, bypassing Red Hat Bugzilla entirely. This could work if we had a whitelist of particular packages for which we should redirect bug reports to particular GitLab instances. (Support for GitHub would be an nice bonus, but I understand the API is probably different.)

**Comment by mcatanzaro** [comment]
BTW this is related to #1196, because only crashes in Fedora flatpaks could be reported to Red Hat Bugzilla, but crashes in other flatpaks are very important to report too. Ideally flatpak crash reports should go straight to the upstream developers.

**Comment by ernestask** [comment]
Well, it would probably be doable with some heuristics to find the upstream issue tracker, but I’m afraid said upstreams would not be too thrilled to get such automated reports. Unless we could extend AppStream metadata to have a field like `accepts_automated_bug_reports` or somesuch.

**Comment by ernestask** [comment]
Or we something rather basic and get the user to input the issue tracker, so it sort of offloads the responsibility from us.

**Comment by ernestask** [comment]
Oh, I missed the whitelist suggestion. Which projects do you have in mind particularly?

**Comment by mcatanzaro** [comment]
> Oh, I missed the whitelist suggestion. Which projects do you have in mind particularly?

I would start with only GNOME projects. If it works out for GNOME, then in the future we could whitelist other projects if requested by the upstream maintainer.

The most likely issue with automatic bug reporting to GNOME GitLab would be if ABRT creates a bunch of duplicate issue reports, like Bug Buddy used to do. That would be likely to result in complaints from maintainers. Now, ABRT seems to be pretty good about flagging potential duplicates on Red Hat Bugzilla, and in avoiding duplicates entirely if the bug reports are all public. Much of the trouble with duplicate reports on Red Hat Bugzilla seems to be caused by ABRT defaulting all bug reports to private nowadays, and intentionally creating duplicate reports of private issues. So I think ABRT is already smart enough, or almost smart enough, to do the right thing with respect to duplicate issues. As long as it avoids reporting separate issues for crashes with very similar backtraces, I expect bug reports would be accepted upstream.

Another potential issue would be creation of crash reports for old software versions. We'd only want to receive crash reports from the latest two major software versions (3.32 and 3.34 currently).

**Comment by ernestask** [comment]
Right, so I imagine it should be good enough as far as supported versions of GNOME go, since the development cycles are aligned.

If we did FAF-initiated reporting based on thresholds (like in the past, IIRC), this could be avoided completely, but I haven’t given it much thought yet.

**Comment by mcatanzaro** [comment]
Oh yeah, sorry. I was thinking about RHEL (where bug reports always belong downstream) and other LTS distributions that don't even use ABRT when I wrote that comment. For Fedora, it already lines up nicely enough that it wouldn't be a problem.

**Comment by ernestask** [comment]
Yeah, I think it’s a bit premature to think about world domination.
